reloc/build_rpm.sh

```diff
@@ -37,7 +37,7 @@ tar -C $BUILDDIR -xpf $RELOC_PKG scylla-gdb/SCYLLA-RELOCATABLE-FILE scylla-gdb/S
 cd $BUILDDIR/scylla-gdb
 
 RELOC_PKG_BASENAME=$(basename "$RELOC_PKG")
-SCYLLA_VERSION=$(cat SCYLLA-VERSION-FILE)
+SCYLLA_VERSION=$(sed 's/-/~/' SCYLLA-VERSION-FILE)
```

Comment:
Doesn't `~` change the meaning?   x.z > x.y, but x~z < x.y.

Reply:
Where do we have '-' in version strings? Is it in the user-provided SCYLLA_BUILD? Let's add a comment and check there instead, to avoid different version strings in different package formats.

Reply:
This is not new code, since it is just copying a code from scylla-core repo, we use same code on all repo.

Anyway, since we moved to [semver-compatible version](https://github.com/scylladb/scylladb/pull/10957), reloc_rpm.sh does not able to build without `scylla_version = file.read().strip().replace('-', '~')`.
Because RPM parses the version string as "[version]-[release]", so  "5.x.y-dev-$RELEASE" is illegal version string since there is additional '-'.
To solve this without stop using semver-compatible versioning, we need to replace "-dev" to "~dev" in build_rpm.sh.